File: JAVA_APPLICATION/app/src/main/java/com/example/myapplication/BookingHistoryAdapter.java
package com.example.myapplication;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import java.util.Map;

public class BookingHistoryAdapter extends RecyclerView.Adapter<BookingHistoryAdapter.BookingViewHolder> {

    private List<Map<String, String>> bookings;

    public BookingHistoryAdapter(List<Map<String, String>> bookings) {
        this.bookings = bookings;
    }

    public static class BookingViewHolder extends RecyclerView.ViewHolder {
        public TextView stationName;
        public TextView startTime;
        public TextView duration;
        public TextView price;

        public BookingViewHolder(View itemView) {
            super(itemView);
            stationName = itemView.findViewById(R.id.textStation);
            startTime = itemView.findViewById(R.id.textStartTime);
            duration = itemView.findViewById(R.id.textDuration);
            price = itemView.findViewById(R.id.textPrice);
        }
    }

    @Override
    public BookingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_booking, parent, false);
        return new BookingViewHolder(view);
    }

    @Override
    public void onBindViewHolder(BookingViewHolder holder, int position) {
        Map<String, String> booking = bookings.get(position);
        holder.stationName.setText("Station: " + booking.get("station"));
        holder.startTime.setText("Start Time: " + booking.get("startTime"));
        holder.duration.setText("Duration: " + booking.get("duration") + " mins");
        holder.price.setText("Price: $" + booking.get("price"));
    }

    @Override
    public int getItemCount() {
        return bookings.size();
    }
}

File: app/src/main/java/com/example/city_cycle_app/BookingHistoryAdapter.kt
package com.example.city_cycle_app

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class BookingHistoryAdapter(private val bookings: List<Map<String, String>>) :
    RecyclerView.Adapter<BookingHistoryAdapter.BookingViewHolder>() {

    class BookingViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val stationName: TextView = itemView.findViewById(R.id.textStation)
        val startTime: TextView = itemView.findViewById(R.id.textStartTime)
        val duration: TextView = itemView.findViewById(R.id.textDuration)
        val price: TextView = itemView.findViewById(R.id.textPrice)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookingViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_booking, parent, false)
        return BookingViewHolder(view)
    }

    override fun onBindViewHolder(holder: BookingViewHolder, position: Int) {
        val booking = bookings[position]
        holder.stationName.text = "Station: ${booking["station"]}"
        holder.startTime.text = "Start Time: ${booking["startTime"]}"
        holder.duration.text = "Duration: ${booking["duration"]} mins"
        holder.price.text = "Price: $${booking["price"]}"
    }

    override fun getItemCount() = bookings.size
}
